### PR TITLE
[AVOCADO-249] Support BP resolution gVCF output.

### LINKED ISSUE #249: Add gVCF output
Splitting out of #199.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #249.

### CODE REVIEW

File: avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/BiallelicGenotyper.scala [hunk 1]
```diff
@@ -313,13 +388,16 @@ private[avocado] object BiallelicGenotyper extends Serializable with Logging {
...
    * @param ploidy The assumed copy number at each site.
+   * @param scoreAllSites If true, calculates likelihoods for all sites, and not
+   *   just variant sites.
    * @return Returns an RDD of (Variant, Observation) pairs.
    */
   private[genotyping] def readsToObservations(
     rdd: RDD[(AlignmentRecord, Iterable[Variant])],
-    ploidy: Int): RDD[(Variant, Observation)] = ObserveReads.time {
+    ploidy: Int,
+    scoreAllSites: Boolean): RDD[(Variant, Observation)] = ObserveReads.time {
```
Comment (heuermh) [inline avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/BiallelicGenotyper.scala]: `scoreAllSites` gets passed around a few times, is it worth creating a scoring strategy or abstract class + subclasses for `BiallelicGenotyper`?

File: avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/SquareOffReferenceModel.scala [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            .getReferenceAllele
+            .dropRight(toTrimFromRight))
+          .setAlternateAllele(gt.getVariant
+            .getAlternateAllele
+            .dropRight(toTrimFromRight))
+          .build)
+      } else {
+        None
+      }
+    })).transformDataset(_.dropDuplicates("start",
```
Comment (heuermh) [inline avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/SquareOffReferenceModel.scala]: Do you want to add doc that this class realizes the RDD as a dataset, or is that expected across Avocado?

File: avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/BiallelicGenotyper.scala (hunk 1 above)
Comment (fnothaft) [reply avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/BiallelicGenotyper.scala]: I'd rather not.

File: avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/SquareOffReferenceModel.scala (hunk 2 above)
Comment (fnothaft) [reply avocado-core/src/main/scala/org/bdgenomics/avocado/genotyping/SquareOffReferenceModel.scala]: There's a mix, but overall the codebase is trending towards SQL.

**Pull request merged**